File: largestpermutation.py
def KswapPermutation(arr, n, k):

    # Auxiliary array of storing the position of elements
    pos = {}
    for i in range(n):
        pos[arr[i]] = i

    for i in range(n):

        # If K is exhausted then break the loop
        if k == 0:
            break

        # If element is already largest then no need to swap
        if (arr[i] == n-i):
            continue

        # Find position of i'th largest value, n-i
        temp = pos[n-i]

        # Swap the elements position
        pos[arr[i]] = pos[n-i]
        pos[n-i] = i

        # Swap the ith largest value with the value at
        # ith place
        arr[temp], arr[i] = arr[i], arr[temp]

        # Decrement K after swap
        k = k-1

# Driver code
num_array = list()
num = raw_input( )
for i in range(int(num)):
    n = input("")
    num_array.append(int(n))
arr = [4, 5, 2, 1, 3]
n = len(arr)
k = 1
KswapPermutation(arr, n, k)

# Print the answer
print "Largest permutation after", k, "swaps: "
print " ".join(map(str,arr))

File: largestpermutation.cpp
// Below is C++ code to print largest permutation
// after atmost K swaps
#include<bits/stdc++.h>
#include<vector>
#include <iostream>
#include <array>
using namespace std;

// Function to calculate largest permutation after
// atmost K swaps
void KswapPermutation(int arr[], int n, int k)
{
    // Auxiliary dictionary of storing the position
    // of elements
    int pos[n+1];

    for (int i = 0; i < n; ++i)
        pos[arr[i]] = i;

    for (int i=0; i<n && k; ++i)
    {
        // If element is already i'th largest,
        // then no need to swap
        if (arr[i] == n-i)
            continue;

        // Find position of i'th largest value, n-i
        int temp = pos[n-i];

        // Swap the elements position
        pos[arr[i]] = pos[n-i];
        pos[n-i] = i;

        // Swap the ith largest value with the
        // current value at ith place
        swap(arr[temp], arr[i]);

        // decrement number of swaps
        --k;
    }
}

// Driver code
int main()
{
    int Nb,k;
    cin >> Nb >> k;
    int* arr=NULL;
  //  cout << sizeof(arr);
    arr=new int[Nb];
    for (int i = 1; i <= Nb; i++) {
      cin >> arr[i] ;
    }
   for (int j = 1; j <= Nb; j++) {
      cout << arr[j] ;
   }
  //  int n = 8/sizeof(arr[0]);
    KswapPermutation(arr,Nb, k);
    cout << "Largest permutation after "
         << k << " swaps:n ";
    for (int z=1; z<=Nb; ++z)
        printf("%d ", arr[z]);
    return 0;
}
